tools/rtpsrelay/control/RtpsRelayControl.cpp

```diff
@@ -214,6 +218,15 @@ int run(int argc, ACE_TCHAR* argv[])
   participant->get_default_subscriber_qos(subscriber_qos);
   subscriber_qos.partition.name = publisher_qos.partition.name;
 
+  DDS::Publisher_var publisher;
+  if (!config.config().empty() || !denied_partitions.empty()) {
+    publisher = participant->create_publisher(publisher_qos, nullptr, 0);
```

Comment:
The denied partitions writer could use a publisher that doesn't have partition QoS so it writes to all instances.